lib/datadog/appsec/security_engine/runner.rb

```diff
@@ -56,6 +57,7 @@ def run(persistent_data, ephemeral_data, timeout = WAF::LibDDWAF::DDWAF_RUN_TIME
 
         def finalize!
           @waf_context.finalize!
+          AppSec.security_engine.release_waf_handle(@waf_handle)
```

Comment:
This will violate separation of concerns. What if waf_handle will be not handle but registry itself?

With that in mind we can acquire and decommission handles?

Reply:
addressed in [761c125](https://github.com/DataDog/dd-trace-rb/pull/4833/commits/761c125cf9cb514600144decd744a24e69e265c0)